NuGet updates fail if repo contains .config/dotnet-tools.json

## Describe the bug

If a repo contains a `.config/dotnet-tools.json` file (like [dotnet-bumper](https://github.com/martincostello/dotnet-bumper/blob/main/.config/dotnet-tools.json)), the reusable workflow fails to restore NuGet packages.

## Steps To reproduce

Run [update-dotnet-sdk.yml](https://github.com/martincostello/update-dotnet-sdk/blob/main/.github/workflows/update-dotnet-sdk.yml) for a repo with a .NET SDK update available with NuGet package updates enabled.

## Expected behaviour

The workflow succeeds with no errors.

## Actual behaviour

The workflow fails with the following:

```console
Installing dotnet-outdated-tool...
You can invoke the tool using the following command: dotnet-outdated
Tool 'dotnet-outdated-tool' (version '4.6.0') was successfully installed.
Restoring NuGet packages...
  Determining projects to restore...
  Restored /home/runner/work/github-automation/github-automation/src/DotNetBumper.TestLogger/DotNetBumper.TestLogger.csproj (in 1.38 sec).
  Restored /home/runner/work/github-automation/github-automation/src/DotNetBumper.MSBuild/DotNetBumper.MSBuild.csproj (in 17.29 sec).
  Restored /home/runner/work/github-automation/github-automation/src/DotNetBumper.Core/DotNetBumper.Core.csproj (in 19.07 sec).
  Restored /home/runner/work/github-automation/github-automation/src/DotNetBumper/DotNetBumper.csproj (in 19.08 sec).
  Restored /home/runner/work/github-automation/github-automation/tests/DotNetBumper.Tests/DotNetBumper.Tests.csproj (in 19.08 sec).
Checking for .NET NuGet package(s) to update...
Run "dotnet tool restore" to make the "dotnet-outdated" command available.
There are no .NET NuGet packages to update.

Error: Process completed with exit code 1.
```

## Additional context

[Failed workflow logs](https://github.com/martincostello/github-automation/actions/runs/8255079111/job/22584120934#step:8:145).

--- a/.github/workflows/update-dotnet-sdk.yml
+++ b/.github/workflows/update-dotnet-sdk.yml
@@ -295,6 +295,14 @@ jobs:
           }
         }
 
+        # See https://github.com/martincostello/update-dotnet-sdk/issues/785
+        $toolsManifest = (Join-Path "." ".config" "dotnet-tools.json")
+
+        if (Test-Path $toolsManifest) {
+          Write-Host "Restoring .NET tools..."
+          dotnet tool restore
+        }
+
         # HACK Inspired by https://github.com/dotnet-outdated/dotnet-outdated/issues/55
         # run dotnet restore to populate the NuGet package cache to try and avoid timeouts
         # when dotnet-outdated is run

